base/multidimensional.jl

```diff
@@ -203,9 +203,7 @@ module IteratorsMD
     CartesianIndex(2, 2, 2)
 
     julia> CartesianIndices(fill(1, (2,3)))
-    2×3 CartesianIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}:
-     CartesianIndex(1, 1)  CartesianIndex(1, 2)  CartesianIndex(1, 3)
-     CartesianIndex(2, 1)  CartesianIndex(2, 2)  CartesianIndex(2, 3)
+    CartesianIndices((Base.OneTo(2), Base.OneTo(3)))
```

Comment:
Not sure if it's worth the trouble, but we might want to even print that as `CartesianIndices((2, 3))` instead.

Reply:
Unfortunately that wouldn't be accepted by the constructor. I like output formats that parse to produce the respective values again.

Reply:
There is actually a corresponding constructor, that's what I meant:
```
julia> CartesianIndices((2, 3))
2×3 CartesianIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}:
 CartesianIndex(1, 1)  …  CartesianIndex(1, 3)
 CartesianIndex(2, 1)     CartesianIndex(2, 3)
```

Reply:
What would the rule be? All `OneTo` values are printed as scalars instead?
```Julia
julia> CartesianIndices((2, 1:3))
CartesianIndices((Base.OneTo(2), 1:3))
```

Reply:
I think the behavior for plain integers mixed with ranges changed at some point, so we might only want to do that if all values are `OneTo`.

Reply:
Done.